### PR TITLE
Various small updates to bar_hotkeys_grid.lua

### PR DESCRIPTION
**Pull request by resopmok**
Changes to grid optimized hotkey layout with reasoning:
- Removed settargetnoground from n. Most players use right click to default for settargetnoground so this frees n for a new function.
- Added hotkeys for newly integrated commandqmanager. n will cancel the current order in a unit's queue (command_skip_current), and ctrl+n will remove the last order from the queue (command_cancel_last).
- Added gatherwait to p. Gatherwait will wait for an entire group of units to arrive before continuing to the next queued waypoint. To use it, set the waypoint for your units with move or fight, press p, then press shift and set the next waypoint.
- Added additional selection key to Ctrl+r which will select all idle transports on the map.
- Added additional selection key to Ctrl+y which will select any waiting units on your screen. This will allow you to easily select any construction turrets or constructors which have been hastily on wait in order to focus on a different eco project.
- States replace some commands with toggles. This will affect onoff (b), firestate (l), movestate (;), and repeat (t). The wait action is currently unable to accept stateful orders and so could not be added. This change sets a state for the toggle using a keychain. For example, a single press of t will set a unit to repeat regardless of its previous state, while a double-tap of t will set a unit to not repeat regardless of its previous state. A single tap of l will set units to fire at will, while a double-tap of l will set units to hold fire, and a triple tap will set them to return fire (yellow). Being able to define the state of the toggle is useful for making sure all units selected have the same toggle behavior. They also allow for group hold fire micro. When using with units that have long reloads, e.g., rocket bots, you would first double-tap l, right-click or use s to set your target, move the units into range of the target, then single-tap l so they fire simultaneously.

### CODE REVIEW

File: luaui/configs/bar_hotkeys_grid.lua [hunk 1]
```diff
@@ -1,6 +1,6 @@
--- BAR hotkey config file: ergokeys for grid menu	
+-- BAR hotkey config file: optimized for use with grid build menu.	
 	local bindings = {
-		{            "esc", "select", "AllMap++_ClearSelection+" },
+		{            "esc", "select", "AllMap++_ClearSelection_SelectNum_0+" },
```
Comment (badosu) [inline luaui/configs/bar_hotkeys_grid.lua:3]: Why is this necessary?

Comment (resopmok) [reply luaui/configs/bar_hotkeys_grid.lua:3]: Yes as I recall I got some errors with it during testing (missing token?) so I thought maybe it was missing the answer to the question, "what do I select?" My thought was to tell it select nothing, and it seemed to fix the error.

Comment (badosu) [reply luaui/configs/bar_hotkeys_grid.lua:3]: Gotcha, I'll keep that in mind, thank you @sprunk

Comment (badosu) [reply luaui/configs/bar_hotkeys_grid.lua:3]: All clear from my end
